src/plugins/es_ui_shared/static/forms/components/form_row.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+
+import { EuiDescribedFormGroup, EuiTitle } from '@elastic/eui';
+import { FieldHook } from '../hook_form_lib';
+import { Field } from './field';
+
+interface Props {
+  title: string | JSX.Element;
+  description?: string | JSX.Element;
+  field?: FieldHook;
+  euiFieldProps?: Record<string, any>;
+  idAria?: string;
+  children?: React.ReactNode;
+  [key: string]: any;
+}
+
+export const FormRow = ({ title, idAria, description, field, children, ...rest }: Props) => {
+  // If a string is provided, create a default Euititle of size "m"
+  const _title =
+    typeof title === 'string' || title.type.name === 'FormattedMessage' ? (
+      <EuiTitle size="s">
+        <h4>{title}</h4>
```

Comment:
Heading hierarchies can't skip levels. e.g. a page contains an `h1` and an `h3` but no `h2`, so if the page doesn't contain `h1` through `h3` this will be flagged as an accessibility bug.

Reply:
This is something important mainly for SEO, I am not sure it has the importance you think it has in Kibana. I audited a few of our forms and realized the rule you mentioned is not followed (no `<h2>` to be found). And I think it's OK as a `<h2>` tag for a form input title seems a lot to me (but I might be wrong).

You are probably right in theory but I am not sure that we need to be too strict about the rule (the same that we won't render `<form>` tags but a `<div>` to wrap our forms).

Also, the component allows a JSX element to be provided if an `<h4>` is not what's desired.